Cannot install HPX in release mode on Rostam

## Expected Behavior
For HPX to build and install

## Actual Behavior
HPX installs without issues in debug mode, but not in release. I am also unable to reproduce this issue anywhere else.

The error:
```
CMake Error at cmake_install.cmake:119 (file):
  file INSTALL cannot find
  "/home/pamini/code/hpx/cmake-build-release/CMakeFiles//hpx_application.pc":
  No such file or directory.


FAILED: CMakeFiles/install.util 
cd /home/pamini/code/hpx/cmake-build-release && /home/pamini/.local/cmake/3.16.4/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed
```


## Steps to Reproduce the Problem

  1. Prepare the environment
```
module purge
module load boost/1.70.0-gcc8.3.0-release gcc/8.3.0 pybind11/2.2.4 mpi/mvapich2-2.2-x86_64 ninja/1.9.0 binutils/2.32 python/3.6.3s blaze/3.4 cmake/3.14.2
```
  1. `git clone --depth=1 --branch=stable https://github.com/stellar-group/hpx.git repo`
  1. `cmake -Hrepo -Bcmake-build-release -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/pamini/.local/hpx/stable-26-g8f5487b-release`
  1. `cmake --build cmake-build-release --target install`

## Specifications

  - HPX Version: 8f5487b4580a804ae0f1d8ee9a6d26aad7c74dd6 (current stable tag)
  - Platform (compiler, OS): Rostam, CentOS, GCC 8.3.0, Boost 1.70.0 Release

## Log
* [Output log](https://github.com/STEllAR-GROUP/hpx/files/4215451/make.log)

--- a/cmake/HPX_GeneratePackageUtils.cmake
+++ b/cmake/HPX_GeneratePackageUtils.cmake
@@ -243,7 +243,7 @@ function(hpx_generate_pkgconfig_from_target target template is_build)
     INPUT ${OUTPUT_DIR}${template}_${build_type}.pc.in
     )
   # Temporary (to deprecate gradually)
-  if ("${CMAKE_BUILD_TYPE}" MATCHES "Rel")
+  if ("${build_type}" MATCHES "rel")
     file(GENERATE
       OUTPUT ${OUTPUT_DIR}/${template}.pc
       INPUT ${OUTPUT_DIR}${template}_${build_type}.pc.in

